code/modules/fishing/fish/fish_traits.dm

```diff
@@ -206,16 +206,23 @@ GLOBAL_LIST_INIT(fish_traits, init_subtypes_w_path_keys(/datum/fish_trait, list(
 	name = "Self-Revival"
 	catalog_description = "This fish shows a peculiar ability of reviving itself a minute or two after death."
 	guaranteed_inheritance_types = list(/obj/item/fish/boned, /obj/item/fish/mastodon)
+	/// Lazylist of string refs keyed to the fish with this trait, with the values being the timer id of the revival timer.
+	var/list/fish_to_timer_ids = null
 
 /datum/fish_trait/revival/apply_to_fish(obj/item/fish/fish)
 	RegisterSignal(fish, COMSIG_FISH_STATUS_CHANGED, PROC_REF(check_status))
 
 /datum/fish_trait/revival/proc/check_status(obj/item/fish/source)
 	SIGNAL_HANDLER
-	if(source.status == FISH_DEAD)
-		addtimer(CALLBACK(src, PROC_REF(revive), source), rand(1 MINUTES, 2 MINUTES))
+	if(source.status != FISH_DEAD)
+		return
+
+	var/timer_id = addtimer(CALLBACK(src, PROC_REF(revive), source), rand(1 MINUTES, 2 MINUTES), flags = TIMER_STOPPABLE)
```

Comment:
why not just use a weakref?

Reply:
i was pondering using that but i figured it wasn't necessary for what we needed to accomplish here (which is just store a list of strings (timer ids), and things to look up those strings)

Reply:
I think they mean "why not pass the fish-to-be-revived via weakref in the callback", I think that would work too

Reply:
oh i see, i think this is probably the most elegant solution